fixtures/flight/src/index.js

```diff
@@ -1,28 +1,28 @@
 import * as React from 'react';
 import {Suspense} from 'react';
 import ReactDOM from 'react-dom/client';
-import ReactServerDOMReader from 'react-server-dom-webpack/client';
+import {createFromFetch, encodeReply} from 'react-server-dom-webpack/client';
 
 // TODO: This should be a dependency of the App but we haven't implemented CSS in Node yet.
 import './style.css';
 
-let data = ReactServerDOMReader.createFromFetch(
+let data = createFromFetch(
   fetch('/', {
     headers: {
       Accept: 'text/x-component',
     },
   }),
   {
-    callServer(id, args) {
+    async callServer(id, args) {
       const response = fetch('/', {
         method: 'POST',
         headers: {
           Accept: 'text/x-component',
           'rsc-action': id,
         },
-        body: JSON.stringify(args),
+        body: await encodeReply(args),
       });
-      return ReactServerDOMReader.createFromFetch(response);
+      return createFromFetch(response);
```

Comment:
Why do you model the resolve handling like this is a RSC render? I get that you may want to return a rendered tree but if this is just a "API" endpoint that returns JSON data or something it seems odd to process the response like it is a component tree. It's interesting that if you did render suspense boundaries you could get an object that could construct itself incrementally as bits stream in but I have a hard time wrapping my head around that as a general purpose response handler instead of something specifc to fetching and integrating flight payloads

Reply:
It's not *just* a renderer. It's a general purpose serializer with much more rich options of the payload, and more to come. So it's used as a better alternative to JSON.stringify.

Since that implementation also happens to be what you'd use to render a real response which likely contains an update to the current page, it's conveniently the same thing.

Reply:
right but if I send `{ eager: "eager", lazy: sleep(5000).then(() => "lazy") }`

i still have to wait 5 seconds to see the result because there is no way to expose an incomplete value and then update it to a complete value. We use Suspense boundaries for React trees and we can re-evaluate the value on each render attempt, but for a general purpose response object we'd need some similar kind of construct which groups a value into revealable. I mean we could literally use suspense but that just feels kind strange

Do you have a vision for what this more generalizable type is so you can really take advantage of the unique characteristics of the flight format and runtime?

Reply:
I think you're just describing Promises? The consumer can just `const value = await result.lazy`?

Reply:
It's not really anything fancy. It's just passing values that are already common place in the language so you just use it as if it was local.

Reply:
So this basically allows us to stream JSON back to the client by serialising some properties as promises, which is pretty cool, I must say. 🎉  Although this is already working in `main` and has nothing to do with this change, but I'm just now realising this. So this example would be serialised to
```
0:"$@1"
1:{"eager":"eager","lazy":"$@2"}
2:"lazy"
```
where the eager chunk is sent immediately, and the lazy chunk is sent after 5 seconds. With async iterators soon to be serialisable, I guess you could use that e.g. to render the progress of a long running task?